### PR TITLE
Moved preallocation and preregistering to render contexts

### PR DESCRIPTION
**Pull request by emilwestergren**
@davidpiuva peer review

### CODE REVIEW

**Comment by davidpiuva** [comment]
ok

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 1]
```diff
@@ -141,5 +141,10 @@ AndroidContext: class extends OpenGLContext {
 		map add("startY", startY)
 		DrawState new(target) setMap(map) draw()
 	}
+	preallocate: override func (resolution: IntVector2D) { GraphicBufferYuv420Semiplanar free~all() }
+	preregister: override func (image: Image) {
+		if (image instanceOf(GraphicBufferYuv420Semiplanar))
+			(image as GraphicBufferYuv420Semiplanar) toRgba(this) referenceCount decrease()
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/AndroidContext.ooc]: You might have to re-test this later since I am not sure if YUV to RGBA works right at this moment.

**Comment by davidpiuva** [comment]
I don't see any obvious mistakes in the code so I am done reviewing.

File: source/draw/gpu/opengl/AndroidContext.ooc (hunk 1 above)
Comment (emilwestergren) [inline source/draw/gpu/opengl/AndroidContext.ooc]: This is actually not a conversion, it's more a type cast. We simply interpret the YUV buffer as an RGBA buffer instead. It is then converted to YUV in the unpack/pack shaders.

Comment (davidpiuva) [inline source/draw/gpu/opengl/AndroidContext.ooc]: But if I flip YVU to YUV or BGR to RGB then the typecast may break from my changes if it depends on order.

**Pull request merged**